scripts/profiler/processed_events.py

```diff
@@ -27,37 +32,127 @@ def __init__(self):
+    @staticmethod
+    def _calculate_md5_hash_of_file(filename):
+        return hashlib.md5(open(filename, 'rb').read()).hexdigest()
+
+    def _init_writing_events_csv(self, filename):
+        try:
+            csvfile = open(filename, 'w', newline='')
+            fieldnames = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
+            wr = csv.DictWriter(csvfile, delimiter=',',
+                                fieldnames=fieldnames)
+            wr.writeheader()
+            return csvfile
+        except IOError:
+            self.logger.error("Problem with accessing file: " + filename)
+            sys.exit()
+
+    def _write_events_csv(self, filename):
+        try:
+            with open(filename, 'w', newline='') as csvfile:
+                fieldnames = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
+                wr = csv.DictWriter(csvfile, delimiter=',',
+                                    fieldnames=fieldnames)
+                wr.writeheader()
+                for ev in self.tracked_events:
+                    wr.writerow({
+                                 'type_id': ev.submit.type_id,
+                                 'timestamp': ev.submit.timestamp,
+                                 'data': ev.submit.data,
+                                 'proc_start_time': ev.proc_start_time,
+                                 'proc_end_time': ev.proc_end_time
+                                })
+        except IOError:
+            self.logger.error("Problem with accessing file: " + filename)
+            sys.exit()
+
+    def _read_events_csv(self, filename):
+        try:
+            with open(filename, 'r', newline='') as csvfile:
+                rd = csv.DictReader(csvfile, delimiter=',')
+                for row in rd:
+                    type_id = int(row['type_id'])
+                    timestamp = float(row['timestamp'])
+                    # reading event data from single row in csv file
+                    if row['data'][1:-1] != '':
+                        data = list(map(lambda s, data_type: s if data_type == "s" else int(s),
+                                        row['data'][1:-1].split(','),
+                                        self.registered_events_types[type_id].data_types))
+                    else:
+                        data = []
+                    proc_start_time = float(row['proc_start_time']) if row['proc_start_time'] != '' else None
+                    proc_end_time = float(row['proc_end_time']) if row['proc_end_time'] != '' else None
+                    ev = TrackedEvent(Event(type_id, timestamp, data), proc_start_time, proc_end_time)
+                    self.tracked_events.append(ev)
+        except IOError:
+            self.logger.error("Problem with accessing file: " + filename)
+            sys.exit()
+
+    def _init_writing_events_types_json(self, filename):
+        d = dict((k, v.serialize())
```

Comment:
Why is writing to json actually defined twice ?

Reply:
Maybe we could only add the `csv_hash` to the data later on, instead of rewriting everything